models/weight_optimizer.h

```diff
@@ -68,21 +68,29 @@ The TensorFlow implementation deviates from [1]_ in that it assumes
 :math:`\hat{\epsilon} = \epsilon \sqrt{ 1 - \beta_2^t }` to be constant, whereas [1]_
 assumes :math:`\epsilon = \hat{\epsilon} \sqrt{ 1 - \beta_2^t }` to be constant.
 
+When `optimize_each_step` is set to `True`, the weights are optimized at every
+time step. If set to `False`, optimization occurs once per spike, resulting in a
+significant speed-up. For gradient descent, both settings yield the same
+results. However, for the Adam optimizer, only setting `optimize_each_step` to
+`True` precisely implements the algorithm as described in [2]_. The impact of
+this setting on learning performance may vary depending on the task.
```

Comment:
```suggestion
results under exact arithmetic; however, small numerical differences may be 
observed due to floating point precision. For the Adam optimizer, only setting
`optimize_each_step` to `True` precisely implements the algorithm as described
in [2]_. The impact of this setting on learning performance may vary depending
on the task.
```
I think this clarification could avoid confusion in case a user starts observing small deviations after several iterations when comparing the 2 options for gradient descent.

Reply:
That's a good point! Is there a word missing after arithmetic?

Reply:
@akorgor no, I don't think so as long as "exact arithmetic" is interpreted as "without any loss of precision or rounding errors"

Reply:
Ah, sorry, I thought it was an adjective and the corresponding noun would be arithmetics - but you are right, the noun is arithmetic.